### PR TITLE
Signing: manifests inside a manifest list should also be signed

### PR DESCRIPTION
**Pull request by vfreex**
When signing a heterogeneous (multi) payload, manifests inside the manifest list should also be signed. This is missed in https://github.com/openshift-eng/aos-cd-jobs/pull/3817.

### CODE REVIEW

File: pyartcd/pyartcd/pipelines/promote.py [hunk 1]
```diff
@@ -472,11 +472,23 @@ async def sign_artifacts(self, release_name: str, client_type: str, release_info
...
                 for release_info in release_infos.values():
+                    version = release_info["metadata"]["version"]
                     pullspec = release_info["image"]
                     digest = release_info["digest"]
+                    json_digests.append((version, pullspec, digest))
+                    # if this payload is a manifest list, iterate through each manifest
+                    manifests = release_info.get("manifests", [])
+                    for manifest in manifests:
+                        if manifest["platform"]["os"] != "linux":
+                            raise ValueError("Unsupported OS %s in manifest list %s", manifest["platform"]["os"], release_info["image"])
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/promote.py:485]: Out of curiosity: Why is this?

**Review by joepvd — APPROVED** [review]
/lgtm

File: pyartcd/pyartcd/pipelines/promote.py (hunk 1 above)
Comment (joepvd) [reply pyartcd/pyartcd/pipelines/promote.py:485]: Maybe worthy of a comment

Comment (vfreex) [reply pyartcd/pyartcd/pipelines/promote.py:485]: This is actually a copy-pasta from https://github.com/openshift-eng/aos-cd-jobs/blob/b0063c829947db4c5950035ee993be603ce4e9c2/pyartcd/pyartcd/pipelines/promote.py#L369.
For now the manifest always has `platform.os==linux`. The check is not necessary but kinda defensive programming.

**Pull request merged**